chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -901,38 +843,32 @@ func (b *BitcoindNotifier) dispatchSpendDetailsManually(op wire.OutPoint,
 				"%v: %v", blockHash, err)
 		}
 
+		// Then, we'll manually go over every transaction in it and
+		// determine whether it spends the outpoint in question.
 		for _, tx := range block.Transactions {
-			for _, in := range tx.TxIn {
-				if in.PreviousOutPoint != op {
+			for i, txIn := range tx.TxIn {
+				if txIn.PreviousOutPoint != op {
 					continue
 				}
 
-				// If this transaction input spends the
-				// outpoint, we'll gather the details of the
-				// spending transaction and dispatch a spend
-				// notification to our clients.
-				relTx := chain.RelevantTx{
-					TxRecord: &wtxmgr.TxRecord{
-						MsgTx:    *tx,
-						Hash:     tx.TxHash(),
-						Received: block.Header.Timestamp,
-					},
-					Block: &wtxmgr.BlockMeta{
-						Block: wtxmgr.Block{
-							Hash:   *blockHash,
-							Height: height,
-						},
-						Time: block.Header.Timestamp,
-					},
+				// If it does, we'll construct its spend details
+				// and hand them over to the TxNotifier so that
+				// it can properly notify its registered
+				// clients.
+				txHash := tx.TxHash()
+				details := &chainntnfs.SpendDetail{
+					SpentOutPoint:     &op,
+					SpenderTxHash:     &txHash,
+					SpendingTx:        tx,
+					SpenderInputIndex: uint32(i),
+					SpendingHeight:    int32(height),
 				}
-
-				select {
-				case b.notificationRegistry <- relTx:
-				case <-b.quit:
-					return ErrChainNotifierShuttingDown
+				err := b.txNotifier.UpdateSpendDetails(
+					op, details,
+				)
+				if err != nil {
```

Comment:
won't this continue to scan blocks after the spend was found? seems like we should return the result of `UpdateSpendDetails` directly

Reply:
A1 catch! Perhaps this was intentional 😉